bbcode-src/index.js

```diff
@@ -16,7 +17,7 @@ const options = {
 };
 
 export const RpNBBCode = (code) =>
-  bbob([preset(), lineBreakPlugin()]).process(code, {
+  bbob([preset(), preserveWhitespace(), lineBreakPlugin()]).process(code, {
```

Comment:
The plugin looks good, but looks like it's missing a boolean check for if the plugin is applied or not?

We can get the site settings in `/assets/javascripts/lib/discourse-markdown/bbcode-plugin.js` and apply it to our custom parser. Then we can probably extend this function to be something like:
```js
const RpNBBCode = (code, opts) => {
  const plugins = [preset()];
  if (opts.preserveWhitespace) {
    plugins.push(preserveWhitespace());
  }
  plugins.push(lineBreakPlugin());
  bbob(plugins).process(code, {
  ...
}
```